tests/switchmap_/server/db/misc/test_rows.py

```
@@ -0,0 +1,660 @@
+                row_value,
+                result[i],
+                f"Mismatch at index {i} for attribute '{attr}'",
+            )
+
+    # ----------------------------------------------------------------------
+    # --- Edge case: None input tests ---
+    # ----------------------------------------------------------------------
+
+    def test_device_edge_none(self):
+        """Edge Case: Test device() with None input."""
+        with self.assertRaises(Exception):
```

Ruff (0.8.2)

`assertRaises(Exception)` should be considered evil

(B017)